File: ddc-sdk/src/test/java/AccountApiTest.java
import com.google.common.collect.HashMultimap;
import com.google.common.collect.LinkedHashMultimap;
import com.google.common.collect.Multimap;
import com.radiance.tonclient.Crypto;
import com.sdyc.ddc.api.account.AccountApi;
import com.sdyc.ddc.bean.AccountInfo;
import com.sdyc.ddc.bean.State;
import org.junit.Test;

import java.math.BigInteger;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by fengtianhua on 2022/7/27.
 */
public class AccountApiTest {
    static AccountApi accountApi = new AccountApi();
    static String address = "0:311efb7e43a1809ef9d26b20dd6150bb3742f7ee58288eeff410b54de3be626d";
    static Crypto.KeyPair keyPair = new Crypto.KeyPair(
            "47797230cfd28f2c5893435e759883c3fa4c3fc05397b3f8ce3612849f9a514f",
            "ae4caeb0532e1dadf95b0ad563f0665d0c2e7940f993ba83dcdb2a11233504fb"
    );

    @Test
    public static void addAccountByPlatform() {
        accountApi.addAccountByPlatform(address,"0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9a","accountName5","5",keyPair);
    }

    @Test
    public static void addBatchAccountByPlatform() {
        List<AccountInfo> accounts = new ArrayList<>();

        AccountInfo accountInfo1 = new AccountInfo();
        accountInfo1.setAccount("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9a");
        accountInfo1.setAccountDID("1001");
        accountInfo1.setAccountName("accountName2");
        accounts.add(accountInfo1);

        AccountInfo accountInfo2 = new AccountInfo();
        accountInfo2.setAccount("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9b");
        accountInfo2.setAccountDID("1002");
        accountInfo2.setAccountName("accountName3");
        accounts.add(accountInfo2);

        accountApi.addBatchAccountByPlatform(address,accounts,keyPair);
    }

    @Test
    public static void setSwitcherStateOfPlatform() {
        accountApi.setSwitcherStateOfPlatform(address, true, keyPair);
    }

    @Test
    public static void addAccountByOperator() {
        accountApi.addAccountByOperator(address, "0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9a", "accountName3", "1003", "1", keyPair);
    }

    @Test
    public static void addBatchAccountByOperator() {
        List<AccountInfo> accounts = new ArrayList<>();

        AccountInfo accountInfo1 = new AccountInfo();
        accountInfo1.setAccount("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9a");
        accountInfo1.setAccountDID("1001");
        accountInfo1.setAccountName("accountName2");
        accounts.add(accountInfo1);

        AccountInfo accountInfo2 = new AccountInfo();
        accountInfo2.setAccount("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9b");
        accountInfo2.setAccountDID("1002");
        accountInfo2.setAccountName("accountName3");
        accounts.add(accountInfo2);

        try {
            accountApi.addBatchAccountByOperator(address, accounts, keyPair);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Test
    public static void updateAccState() {
        accountApi.updateAccState(address, "0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9b", State.Frozen, true, keyPair);
    }

    @Test
    public static void recharge() {
        try {
            accountApi.recharge(address, "0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9b", BigInteger.valueOf(1), keyPair);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Test
    public static void rechargeBatch() {
        Multimap<String, BigInteger> list = LinkedHashMultimap.create();
        list.put("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9b", BigInteger.valueOf(1));
        list.put("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9c", BigInteger.valueOf(1));
        list.put("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9d", BigInteger.valueOf(1));
        list.put("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9e", BigInteger.valueOf(1));
        try {
            accountApi.rechargeBatch(address, list, keyPair);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Test
    public static void balanceOf() {
        accountApi.balanceOf("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9b");
    }

    @Test
    public static void balanceOfBatch() {
        List<String> list = new ArrayList<>();
        list.add("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9b");
        list.add("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9c");
        list.add("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9d");
        list.add("0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9e");
        accountApi.balanceOfBatch(list);
    }

    @Test
    public static void selfRecharge() {
        try {
            accountApi.selfRecharge(address, BigInteger.valueOf(1), keyPair);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Test
    public static void setApprovalForAll() {
        try {
            accountApi.setApprovalForAll(address, "0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9c", true, address, keyPair);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Test
    public static void isApprovedForAll() {
        accountApi.isApprovedForAll(address, "0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9c");
    }

    @Test
    public static void safeMultisigWallet() {
        try {
            accountApi.safeMultisigWallet(address, keyPair);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Test
    public static void setTransable() {
        try {
            accountApi.setTransable(address, "0:7d46a1ff2ccf9a140bf21d1baa4a857fc02564e1cdd0095b3cfa51bd624e9d9c", true, keyPair);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

}

File: resources/AccountApiTest.php
<?php 
    include("java/Java.inc");
    $opAddress = "0:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx";
    $opKeyPair = new Java('com.radiance.tonclient.Crypto$KeyPair', 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx');

    $plAddress = "0:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx";
    $plKeyPair = new Java('com.radiance.tonclient.Crypto$KeyPair', 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx');

    $coAddress = "0:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx";
    $coKeyPair = new Java('com.radiance.tonclient.Crypto$KeyPair', 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx');

    // 实例化AccountApi
    $AccountApi = new Java('com.sdyc.ddc.api.account.AccountApi');

    // 运营方添加业务账号
    // $account = "0:4570679b2879bc821cab5dc6439cd12319e1dece2a99b7bdb4331c6f501ab84d";
    // $accountDID = "000002";
    // $leaderDID = "";
    // echo $AccountApi->addAccountByOperator($opAddress, $account, $accountDID, $leaderDID, $opKeyPair)."<BR>";

    // 平台方添加业务账号
    // $account = "0:f2b1993c754a5b1a6d86426e70e77a352b1bf8d71537ef416630a8a24d50dd13";
    // $accountDID = "000003";
    // $leaderDID = "000002";
    // echo $AccountApi->addAccountByPlatform($plAddress, $account, $accountDID, $leaderDID, $plKeyPair)."<BR>";

    // 运营方充值
    // echo $AccountApi->selfRecharge($opAddress, new Java('java.math.BigInteger', 10000000), $opKeyPair);

    // 运营方转账
    // echo $AccountApi->recharge($opAddress, $plAddress, new Java('java.math.BigInteger', 1000), $opKeyPair)."<BR>";

    // 查询余额
    // echo $AccountApi->balanceOf($plAddress)."<BR>";

    // 平台链账户开关查询
    // echo $AccountApi-> switcherStateOfPlatform();

    // 平台链账户开关设置
    // echo $AccountApi->setSwitcherStateOfPlatform($opAddress, true, $opKeyPair);

    // 获取业务账号
    // echo $AccountApi->getAccountAddress($plAddress);

    // 账号状态
    // echo $AccountApi->accountAvailable($AccountApi->getAccountAddress($plAddress));

    // 修改账号状态
    // $state = new Java('com.sdyc.ddc.bean.State');
    // echo $AccountApi->updateAccState($opAddress, $plAddress, $state->valueOf('Frozen'), false, $opKeyPair);

    // 账号信息
    // echo $AccountApi->getAccount($plAddress)->toString();

    // 账号赋权信息
    // echo $AccountApi->isApprovedForAll($coAddress, $plAddress);

    // 账号赋权
    // echo $AccountApi->setApprovalForAll($coAddress, $plAddress, true, $plAddress, $plKeyPair);

    // 删除业务账号
    // echo $AccountApi->delAccount($opAddress, $opKeyPair, true, $coAddress);

    // 创建链账号
    // echo $AccountApi-> safeMultisigWallet($opAddress, $opKeyPair);

    // 离线生成链账号
    // echo $AccountApi-> getSafeMultisigWallet();

    // 部署链账号
    // $deployKeyPair = new Java('com.radiance.tonclient.Crypto$KeyPair', 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx');
    // echo $AccountApi-> deploySafeMultisigWallet($opAddress, $deployKeyPair, $opKeyPair);
    
    // 转账白名单
    // echo $AccountApi-> setTransable($opAddress, $plAddress, true, $opKeyPair);
?> 
